cdn/snippets.py

```diff
@@ -23,199 +23,195 @@
-            expires=expiration_timestamp,
-            key_name=key_name)
+    digest = hmac.new(decoded_key, url_to_sign.encode("utf-8"), hashlib.sha1).digest()
+    signature = base64.urlsafe_b64encode(digest).decode("utf-8")
 
-    digest = hmac.new(
-        decoded_key, url_to_sign.encode('utf-8'), hashlib.sha1).digest()
-    signature = base64.urlsafe_b64encode(digest).decode('utf-8')
+    return f"{url_to_sign}&Signature={signature}"
 
-    signed_url = u'{url}&Signature={signature}'.format(
-            url=url_to_sign, signature=signature)
 
-    print(signed_url)
 # [END cloudcdn_sign_url]
 
 
 # [START cloudcdn_sign_url_prefix]
-def sign_url_prefix(url, url_prefix, key_name, base64_key, expiration_time):
+def sign_url_prefix(
+    url: str,
+    url_prefix: str,
+    key_name: str,
+    base64_key: str,
+    expiration_time: datetime,
+) -> str:
     """Gets the Signed URL string for the specified URL prefix and configuration.
 
     Args:
         url: URL of request.
-        url_prefix: URL prefix to sign as a string.
-        key_name: name of the signing key as a string.
+        url_prefix: URL prefix to sign.
+        key_name: name of the signing key.
         base64_key: signing key as a base64 encoded string.
-        expiration_time: expiration time as a UTC datetime object.
+        expiration_time: expiration time.
 
     Returns:
         Returns the Signed URL appended with the query parameters based on the
         specified URL prefix and configuration.
     """
     stripped_url = url.strip()
-    parsed_url = urllib.parse.urlsplit(stripped_url)
-    query_params = urllib.parse.parse_qs(
-        parsed_url.query, keep_blank_values=True)
+    parsed_url = urlsplit(stripped_url)
+    query_params = parse_qs(parsed_url.query, keep_blank_values=True)
     encoded_url_prefix = base64.urlsafe_b64encode(
-            url_prefix.strip().encode('utf-8')).decode('utf-8')
-    epoch = datetime.datetime.utcfromtimestamp(0)
+        url_prefix.strip().encode("utf-8")
+    ).decode("utf-8")
+    epoch = datetime.utcfromtimestamp(0)
     expiration_timestamp = int((expiration_time - epoch).total_seconds())
     decoded_key = base64.urlsafe_b64decode(base64_key)
 
-    policy_pattern = u'URLPrefix={encoded_url_prefix}&Expires={expires}&KeyName={key_name}'
-    policy = policy_pattern.format(
-            encoded_url_prefix=encoded_url_prefix,
-            expires=expiration_timestamp,
-            key_name=key_name)
+    policy = f"URLPrefix={encoded_url_prefix}&Expires={expiration_timestamp}&KeyName={key_name}"
+
+    digest = hmac.new(decoded_key, policy.encode("utf-8"), hashlib.sha1).digest()
+    signature = base64.urlsafe_b64encode(digest).decode("utf-8")
 
-    digest = hmac.new(
-            decoded_key, policy.encode('utf-8'), hashlib.sha1).digest()
-    signature = base64.urlsafe_b64encode(digest).decode('utf-8')
+    return f"{stripped_url}{'&' if query_params else '?'}{policy}&Signature={signature}"
 
```

Comment:
ditto: please remove empty lines before the closure of the region tag if `black` formatter doesn't enforce them.